api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/FaltososLevantamentoARVCohortQueries.java

```diff
@@ -861,24 +891,45 @@ public CohortDefinition getPatientsMarkedAsBreastfeeding() {
     Map<String, Integer> map = new HashMap<>();
     map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
     map.put("6332", hivMetadata.getBreastfeeding().getConceptId());
+    map.put("1982", hivMetadata.getPregnantConcept().getConceptId());
     map.put("1065", hivMetadata.getPatientFoundYesConcept().getConceptId());
 
     String query =
-        "SELECT p.patient_id "
-            + "FROM   patient p "
-            + "       INNER JOIN encounter e ON e.patient_id = p.patient_id "
-            + "       INNER JOIN obs o ON o.encounter_id = e.encounter_id "
-            + "       INNER JOIN person ps ON ps.person_id = p.patient_id "
-            + "WHERE  e.encounter_type = ${6} "
-            + "       AND e.encounter_datetime BETWEEN Date_add(Curdate(), INTERVAL -18 MONTH) AND Curdate() "
-            + "AND e.location_id = :location"
-            + "       AND e.voided = 0 "
-            + "       AND o.concept_id = ${6332} "
-            + "       AND o.value_coded = ${1065} "
-            + "       AND o.voided = 0 "
-            + "       AND ps.gender = 'F' "
-            + "       AND ps.voided = 0 "
-            + "GROUP BY p.patient_id ";
+
+              "SELECT breastfeeding.patient_id "
+            + "FROM  (SELECT p.patient_id, MAX(e.encounter_datetime) breastfeed_date "
+            + "       FROM   patient p INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + "              INNER JOIN obs o ON o.encounter_id = e.encounter_id "
+            + "              INNER JOIN person ps ON ps.person_id = p.patient_id "
+            + "       WHERE  e.encounter_type = ${6} "
+            + "              AND e.encounter_datetime BETWEEN DATE_ADD(CURDATE(), INTERVAL -18 MONTH) AND CURDATE() "
+            + "              AND e.location_id = :location "
+            + "              AND e.voided = 0 "
+            + "              AND o.concept_id = ${6332} "
+            + "              AND o.value_coded = ${1065} "
+            + "              AND o.voided = 0 "
+            + "              AND ps.gender = 'F' "
+            + "              AND ps.voided = 0 "
+            + "       GROUP  BY p.patient_id) breastfeeding "
+            + "      LEFT JOIN (SELECT p.patient_id, MAX(e.encounter_datetime) pregnancy_date "
+            + "                 FROM   patient p "
+            + "                        INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + "                        INNER JOIN obs o ON o.encounter_id = e.encounter_id "
+            + "                        INNER JOIN person ps ON ps.person_id = p.patient_id "
+            + "                 WHERE  e.encounter_type = ${6} "
+            + "                        AND e.encounter_datetime BETWEEN DATE_ADD(CURDATE(), INTERVAL -9 month) AND CURDATE() "
```

Comment:
is this date same not suppossed to be  end date? if it is now and say someone want a report for 2005 for instance, can we still base on today's date?

Reply:
According to the spec is the report generation date, not end date.

"verify if it falls within the the last 18 months (the max (encounter_datetime) is >= report generation date minus 18 months and <=report generation date)"

Reply:
Ok